app/src/main/java/ml/docilealligator/infinityforreddit/RedditDataRoomDatabase.java

```diff
@@ -40,7 +40,7 @@
         SubscribedUserData.class, MultiReddit.class, CustomTheme.class, RecentSearchQuery.class,
         ReadPost.class, PostFilter.class, PostFilterUsage.class, AnonymousMultiredditSubreddit.class}, version = 23)
 public abstract class RedditDataRoomDatabase extends RoomDatabase {
-    private static RedditDataRoomDatabase INSTANCE;
+    private static volatile RedditDataRoomDatabase INSTANCE;
 
     public static RedditDataRoomDatabase getDatabase(final Context context) {
```

Comment:
By the way double-check here is redundant because singletons in Dagger already use double-check providers

Reply:
Indeed. Replaced DCL with a simple factory method